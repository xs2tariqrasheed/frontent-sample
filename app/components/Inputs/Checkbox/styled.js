import styled from 'styled-components';
import theme from '../../../styles/theme';

export const Wrapper = styled.label`
  position: relative;
  padding-left: 1.8rem;
  padding-top: 0rem;
  cursor: pointer;
  text-align: left;
  font-family: Roboto;
  font-size: 18px;
  margin-bottom: 1rem;
  color: white;

  input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
    height: 0;
    width: 0;

    &:checked ~ span {
      background-color: ${theme.color.lightBackground};

      &:after {
        display: block;
      }
    }
  }

  span {
    position: absolute;
    top: 0;
    left: 0;
    height: 1.4rem;
    width: 1.4rem;
    background: white;
    border: 0.1rem solid ${theme.color.border};
    border-radius: ${theme.borderRadius};

    &:after {
      content: '';
      position: absolute;
      display: none;

      left: 0.35rem;
      top: 0.3rem;
      width: 0.4rem;
      height: 0.7rem;
      border: solid ${theme.color.primary};
      border-width: 0 0.2rem 0.2rem 0;
      transform: rotate(45deg);
    }
  }

  &:hover {
    input ~ span {
    }
  }
`;
