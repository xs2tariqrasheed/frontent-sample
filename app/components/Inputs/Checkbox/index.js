import React from 'react';
import PropTypes from 'prop-types';
import { Wrapper } from './styled';

const Checkbox = ({ label, ...rest }) => (
  <Wrapper>
    {typeof label === 'string' ? label : ''}
    <input type="checkbox" {...rest} />
    <span />
  </Wrapper>
);

Checkbox.propTypes = {
  label: PropTypes.string,
};

export default Checkbox;
