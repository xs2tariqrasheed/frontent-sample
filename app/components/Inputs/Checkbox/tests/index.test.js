import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';

import Checkbox from '../index';

describe('<Checkbox />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <Checkbox />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
