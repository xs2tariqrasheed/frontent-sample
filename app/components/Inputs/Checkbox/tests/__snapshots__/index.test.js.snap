// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render and match the snapshot 1`] = `
<label
  class="styled__Wrapper-n8g513-0 iBZorR"
>
  
  <input
    type="checkbox"
  />
  <span />
</label>
`;
